components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -722,70 +716,72 @@ func SetContainerImageFromRegistry(ctx context.Context, config *rest.Config, not
 						return fmt.Errorf("invalid image selection format")
 					}
 
-					// Specify the GroupVersionResource for imagestreams
-					ims := schema.GroupVersionResource{
-						Group:    "image.openshift.io",
-						Version:  "v1",
-						Resource: "imagestreams",
-					}
-
 					imagestreamFound := false
-					// List imagestreams in the specified namespace
-					imagestreams, err := dynamicClient.Resource(ims).Namespace(namespace).List(ctx, metav1.ListOptions{})
-					if err != nil {
-						if k8serr.IsForbidden(err) {
-							log.Info("Permission denied to list imagestreams", "namespace", namespace, "error", err)
-							// fast exit on permission denied
-							return err
+					imageTagFound := false
+					imagestreamName := imageSelected[0]
+					imgSelection := &imagev1.ImageStream{}
+
+					// Search for the ImageStream in the controller namespace first
+					// As default, the ImageStream is created in the controller namespace
+					// if not found, search in the notebook namespace
+					// Note: This is in this order, so users should not overwrite the ImageStream
+					err := cli.Get(ctx, types.NamespacedName{Name: imagestreamName, Namespace: controllerNamespace}, imgSelection)
+					if err != nil && apierrs.IsNotFound(err) {
```

Comment:
I think we're missing the:
```
span.AddEvent(IMAGE_STREAM_NOT_FOUND_EVENT)
```
for the test. AFAIK this block is a good place to put it in.

Reply:
Thanks @jstourac 
moving this work ahead, will take care of the rest with this: https://issues.redhat.com/browse/RHOAIENG-25707

Reply:
For this one specific item: #606.